mesonbuild/backend/ninjabackend.py

```diff
@@ -972,17 +972,17 @@ def determine_dep_vapis(self, target):
         the same name as the BuildTarget and return the path to it relative to
         the build directory.
         """
-        result = []
+        result = set()
```

Comment:
This potentially changes the order which might cause reproducibility issues. Use an OrderedSet from mesonlib instead.